### PR TITLE
build(go-control-plane): bumped version --> 0.10.1

### PR DESCRIPTION
**Pull request by gunnar-solo**
# Description

Please include a summary of the changes.

This bug fixes ... \ This new feature can be used to ...

# Context

Users ran into this bug doing ... \ Users needed this feature to ...

# Checklist:

- [ ] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [ ] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [ ] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [ ] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -10,15 +10,14 @@ require (
...
-	github.com/cncf/udpa/go v0.0.0-20201120205902-5459f2c99403
 	github.com/containerd/cgroups v0.0.0-20210114181951-8a68de567b68 // indirect
 	github.com/containerd/containerd v1.4.11 // indirect
 	github.com/containerd/continuity v0.0.0-20210208174643-50096c924a4e // indirect
 	github.com/cratonica/2goarray v0.0.0-20190331194516-514510793eaa
 	github.com/docker/cli v20.10.10+incompatible // indirect
 	github.com/docker/docker v20.10.3+incompatible // indirect
 	github.com/elazarl/goproxy v0.0.0-20210110162100-a92cc753f88e // indirect
-	github.com/envoyproxy/go-control-plane v0.9.9-0.20210512163311-63b5d3c536b0
+	github.com/envoyproxy/go-control-plane v0.10.1
```
Comment (gunnar-solo) [inline go.mod:20]: from `go get github.com/envoyproxy/go-control-plane@v0.10.1`

File: hack/filter_types/filter_types.gen.go [hunk 2]
```diff
@@ -15,60 +15,52 @@
 package filter_types
 
 import (
+	_ "github.com/envoyproxy/go-control-plane/contrib/envoy/extensions/filters/http/squash/v3"
```
Comment (gunnar-solo) [inline hack/filter_types/filter_types.gen.go:18]: all changes to this file from `make generated-code -B`

File: projects/gloo/cli/pkg/xdsinspection/get_last_config.go [hunk 3]
```diff
@@ -148,10 +147,10 @@ func getXdsDump(ctx context.Context, xdsPort, proxyName, proxyNamespace string)
 						if err = ptypes.UnmarshalAny(config.TypedConfig, &hcm); err == nil {
 							hcms = append(hcms, hcm)
 						}
-					case *envoylistener.Filter_HiddenEnvoyDeprecatedConfig:
-						if err = envoyutil.StructToMessage(config.HiddenEnvoyDeprecatedConfig, &hcm); err == nil {
-							hcms = append(hcms, hcm)
-						}
+						// case *envoylistener.Filter_HiddenEnvoyDeprecatedConfig:
```
Comment (gunnar-solo) [inline projects/gloo/cli/pkg/xdsinspection/get_last_config.go:150]: No idea what this case is, or what the switch is for!  The type doesn't appear in the newer `go-control-plane` version that I could find, so I killed it.  Hoping for more knowledgable eyes to offer insight

File: projects/gloo/pkg/utils/any.go [hunk 4]
```diff
@@ -51,34 +49,6 @@ func MustAnyToMessage(a *pany.Any) proto.Message {
 	return x.Message
 }
 
-// gogoprotos converted directly to goproto any can't be marshalled unless you wrap
-// the contents of the gogoproto in a typed struct
-func MustGogoMessageToAnyGoProto(msg proto.Message) *pany.Any {
```
Comment (gunnar-solo) [inline projects/gloo/pkg/utils/any.go:56]: There was a proto error with the `GogoMessageToAnyGoProto` after the upgrade.

Seeing as to these two functions didn't seem to be _used_ anywhere in the `gloo` repo, I removed them.

File: projects/sds/pkg/server/server.go [hunk 5]
```diff
@@ -124,7 +124,7 @@ func (s *Server) UpdateSDSConfig(ctx context.Context) error {
 
 	secretSnapshot := cache.Snapshot{}
 	secretSnapshot.Resources[cache_types.Secret] = cache.NewResources(snapshotVersion, items)
-	return s.snapshotCache.SetSnapshot(s.sdsClient, secretSnapshot)
+	return s.snapshotCache.SetSnapshot(ctx, s.sdsClient, secretSnapshot)
```
Comment (gunnar-solo) [inline projects/sds/pkg/server/server.go:127]: Blindly passed `ctx`, here.  The new function header called for an object of that type, and I've seen `ctx` passed a a first parameter a lot in this repo

**Review by kdorosh — APPROVED** [review]
nice! 🎉

**Pull request merged**